### PR TITLE
Separate between class-based, function-based, and const-based use statements by a single blank line #4609

### PR DESCRIPTION
**Pull request by junichi11**
- https://github.com/apache/netbeans/issues/4609
- PSR-12: https://www.php-fig.org/psr/psr-12/#3-declare-statements-namespace-and-import-statements
- Fix the formatter
- Fix the preview file

nb-php-gh-4609-formatter

- Fix "Fix imports"
- Add `FixImportsBottomPanel` as a bottom panel of `FixDuplicateImportsStmts`

Example:
```php
<?php
class TestClass implements MyInterface {
    use MyTrait;
    public function test(): void {
        $a = new MyClass();
        CONSTANT_B;
    }
}
CONSTANT_A;
MyEnum::Case1;
functionA();
```
Inserted (Checked "Put in PSR-12 order"):
```php
<?php
namespace Test;

use Vendor\Package\{
    Classes\MyClass,
    Enums\MyEnum,
    Interfaces\MyInterface,
    Traits\MyTrait
};

use function Vendor\Package\Functions\functionA;

use const Vendor\Package\Constants\{
    CONSTANT_A,
    CONSTANT_B
};

class TestClass implements MyInterface {
    use MyTrait;
    public function test(): void {
        $a = new MyClass();
        CONSTANT_B;
    }
}
CONSTANT_A;
MyEnum::Case1;
functionA();
```

Before:
nb-php-gh-4609-fix-imports-before

After:
nb-php-gh-4609-fix-imports-after

- Don't add a space before `{`
- Don't generate full class names in the panel

#### NOTE
The following is not supported yet.

```
Compound namespaces with a depth of more than two MUST NOT be used. Therefore the following is the maximum compounding depth allowed:
```
```php
<?php

use Vendor\Package\SomeNamespace\{
    SubnamespaceOne\ClassA,
    SubnamespaceOne\ClassB,
    SubnamespaceTwo\ClassY,
    ClassZ,
};
```

### CODE REVIEW

File: php/php.editor/src/org/netbeans/modules/php/editor/actions/FixDuplicateImportStmts.java [hunk 1]
```diff
@@ -121,12 +117,10 @@ public void focusLost(FocusEvent arg0) {
         // load localized text into widgets:
         lblTitle.setText(getBundleString("FixDupImportStmts_IntroLbl")); //NOI18N
         lblHeader.setText(getBundleString("FixDupImportStmts_Header")); //NOI18N
-
-        checkUnusedImports = new JCheckBox();
-        Mnemonics.setLocalizedText(checkUnusedImports, getBundleString("FixDupImportStmts_UnusedImports")); //NOI18N
-        bottomPanel.add(checkUnusedImports, BorderLayout.WEST);
-        checkUnusedImports.setEnabled(true);
-        checkUnusedImports.setSelected(removeUnusedImports);
+        fixImportsBottomPanel = new FixImportsBottomPanel(combos, importData);
```
Comment (junichi11) [inline php/php.editor/src/org/netbeans/modules/php/editor/actions/FixDuplicateImportStmts.java:120]: I've moved the checkbox and buttons of the bottom panel to the new panel to fix them more easily.

Comment (tmysik) [reply php/php.editor/src/org/netbeans/modules/php/editor/actions/FixDuplicateImportStmts.java:120]: Makes sense. We can more easily change it, if needed (e.g. more checkboxes to be added).

File: php/php.editor/src/org/netbeans/modules/php/editor/actions/FixDuplicateImportStmts.form [hunk 2]
```diff
@@ -31,9 +31,10 @@
   </Properties>
   <AuxValues>
     <AuxValue name="FormSettings_autoSetComponentName" type="java.lang.Boolean" value="false"/>
-    <AuxValue name="FormSettings_generateFQN" type="java.lang.Boolean" value="true"/>
+    <AuxValue name="FormSettings_generateFQN" type="java.lang.Boolean" value="false"/>
```
Comment (tmysik) [inline php/php.editor/src/org/netbeans/modules/php/editor/actions/FixDuplicateImportStmts.form:34]: We are changing the default value here - no problem, I think. If there are any complaints, we could store it in the userdir so it is kept between individual invocations (if we are not doing it already :)

File: php/php.editor/src/org/netbeans/modules/php/editor/actions/FixImportsBottomPanel.java [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+import org.openide.util.NbBundle;
+
+class FixImportsBottomPanel extends JPanel {
+
+    private final JComboBox[] comboBoxes;
+    private final ImportData importData;
+    private ImportData.ItemVariant[] defaultVariants;
+    private ImportData.ItemVariant[] dontUseVariants;
+    private static final long serialVersionUID = 7771810767310773171L;
+
```
Comment (tmysik) [inline php/php.editor/src/org/netbeans/modules/php/editor/actions/FixImportsBottomPanel.java:39]: Nitpick: a static field, should be before the instance ones.

**Review by tmysik — APPROVED** [review]
Nice 👍

**Comment by tmysik** [comment]
@junichi11 I am merging, my comments are just nitpicks :)

**Comment by junichi11** [comment]
@tmysik I'll fix it when I change that file next time :) Thank you, Tomas!

**Comment by tmysik** [comment]
@junichi11 Nothing to fix and nothing to thank for, of course :)

**Pull request merged**